libraries/grpc-sdk/src/utilities/StateManager.ts

```diff
@@ -1,11 +1,56 @@
 import { RedisManager } from './RedisManager';
 import { Cluster, Redis } from 'ioredis';
+import Redlock, { Lock } from 'redlock';
+import { Indexable } from '../interfaces';
+
+export enum KNOWN_LOCKS {
+  STATE_MODIFICATION = 'state_modification',
+}
 
 export class StateManager {
   private readonly redisClient: Redis | Cluster;
+  private readonly redLock: Redlock;
 
   constructor(redisManager: RedisManager, name: string) {
     this.redisClient = redisManager.getClient({ keyPrefix: name + '_' });
+    this.redLock = new Redlock([this.redisClient], {
+      // The expected clock drift; for more details see:
+      // http://redis.io/topics/distlock
+      driftFactor: 0.01, // multiplied by lock ttl to determine drift time
+      // The max number of times Redlock will attempt to lock a resource
+      // before error.
+      retryCount: 10,
+
+      // the time in ms between attempts
+      retryDelay: 100, // time in ms
+      // the max time in ms randomly added to retries
+      // to improve performance under high contention
+      // see https://www.awsarchitectureblog.com/2015/03/backoff.html
+      retryJitter: 200, // time in ms
+      // The minimum remaining time on a lock before an extension is automatically
+      // attempted with the `using` API.
+      automaticExtensionThreshold: 500, // time in ms
+    });
+  }
+
+  async acquireLock(resource: string, ttl: number = 5000): Promise<Lock> {
```

Comment:
`resource` type should probably be `KNOWN_LOCKS`

Reply:
No, known locks is used for ease, you can acquire lock in whatever you want